app/models/mixins/retirement_mixin.rb

```diff
@@ -106,8 +106,7 @@ def raise_retire_audit_event(message)
 
   def retirement_check
     return if retired? || retiring? || retirement_initialized?
-    requester = system_context_requester
-
+    requester = system_context_requester ? system_context_requester.userid : system_context_requester
```

Comment:
Should this be more like
`requester = system_context_requester&.userid` handling `nil`
or
`requester = system_context_requester.respond_to?(:userid) ? system_context_requester.userid : system_context_requester` handling string vs user object
?